calico/acl_manager/test/test_acl_store.py

```diff
@@ -95,5 +100,21 @@ def test_case2(self):
         self.acl_pub.test_query_endpoint_acls('e5')
         self.acl_pub.test_wait_assert_all_acls_received()
 
+    def test_case3(self):
+        """
+        Clean shutdown of ACL Manager on ACL Store worker thread crash
+        """
+        # Mock the terminate function so the UT process doesn't exit
+        utils.terminate = Mock()
+
+        self.processor.test_update_endpoint_acls('e1', self.acls)
+        self.acl_pub.test_set_expected_acls('e1', self.acls)
+        self.acl_pub.test_raise_exception()
+
+        # Allow a second for the worker thread to call the mocked terminate
+        time.sleep(1)
```

Comment:
This isn't a great option, because it makes the tests timing sensitive. Consider using the [`event` primitive](https://docs.python.org/2/library/threading.html#event-objects) instead.